skyve-war/src/main/java/modules/admin/ChangePassword/actions/MakePasswordChange.java

```diff
@@ -47,11 +48,28 @@ public ServerSideActionResult<ChangePassword> execute(ChangePassword bean, WebCo
 			throw new ValidationException(message);
 		}
 
-		if (! newPassword.equals(confirmPassword)) { // these 2 are mandatory in the document
+		if (configuration.isHibpEnabled()) {
+			try {
+				// validate if password is PWNED
+				if (HIBPPasswordValidator.isPasswordPwned(newPassword)) {
+					Message message = new Message(ChangePassword.newPasswordPropertyName,
+							"The password you have entered has been breached. "
+									+ "Please re-enter and confirm the password.");
+
+					throw new ValidationException(message);
```

Comment:
Is there a way we can implement this as a warning as per the Trello ticket? If this is enabled, this will prevent someone from using a pwned password at all won't it?